lib/Mojo/mysql/Results.pm

```diff
@@ -18,6 +18,24 @@ sub hash { shift->sth->fetchrow_hashref }
 
 sub hashes { Mojo::Collection->new(@{shift->sth->fetchall_arrayref({})}) }
 
+sub map {
+  return { map @$_, @{ $_[0]->arrays } };
+}
+
+sub map_hashes {
+  my ($self, $keyname) = @_;
+  my $rows = $self->hashes;
+  my @keys;
+
+  $rows->map(sub {
+    push @keys, $_->{ $keyname };
+  });
+
+  my %return;
+  @return{@keys} = @$rows;
+  return \%return;
+}
```

Comment:
This is a lot of code. I would much rather run the code below, without the need of an extra method in Mojo::mysql::Results:

    $mapped = { $res->hashes->map(sub { ($_->{id} => $_) })->each };